indico/modules/events/management/controllers/settings.py

```diff
@@ -145,9 +148,20 @@ def render_form(self, form):
 
 
 class RHEditEventClassification(RHEditEventDataBase):
-    form_class = EventClassificationForm
     section_name = 'classification'
 
+    @property
+    def form_class(self):
+        if allowed_keywords := event_settings.get('allowed_keywords'):
+            choices = [{'id': kw, 'name': kw} for kw in (set(allowed_keywords) | set(self.event.keywords))]
+            keywords = IndicoStrictKeywordsField(_('Keywords'), choices=choices)
+        else:
+            keywords = IndicoTagListField(_('Keywords'))
+        return type('_EventClassificationForm', (EventClassificationForm,), {'keywords': keywords})
+
+    def render_form(self, form):
+        return jsonify_form(form, footer_align_right=True, disabled_until_change=False)
```

Comment:
This shouldn't be necessary since the change event is triggered by the react component and there's no reason to save that form without having made some changes.

```suggestion
```

Reply:
That's still a change which will enable the submit button anyway :)

Not to mention that with the other suggestion this cannot happen unless someone removes the keyword from the whitelist *while you have the edit form open* to add it. That's so unlikely that it's not worth adding extra code for :)

Reply:
Alright! Fixed